ush/python/pygfs/task/snowens_analysis.py

```diff
@@ -2,429 +2,435 @@
-            # set the mask to 3 on glaciers
-            glacier = 15
-            slmsk[np.where(vtype == glacier)] = 3
-            # write out and close the file
-            rst.variables['slmsk'][:] = slmsk
-            rst.close()
+        Analysis: parent class for GDAS task
 
-    @logit(logger)
-    def regridDetBkg(self) -> None:
-        """Run fregrid to regrid the deterministic snow background
-        to the ensemble resolution
-
-        Parameters
+        Returns
         ----------
-        self : Analysis
-           Instance of the SnowEnsAnalysis object
+        None
         """
 
-        chdir(self.task_config.DATA)
-
-        arg_list = [
-            "--input_mosaic", f"./orog/det/{self.task_config.CASE}_mosaic.nc",
-            "--input_dir", f"./bkg/det/",
-            "--input_file", f"{to_fv3time(self.task_config.bkg_time)}.sfc_data",
-            "--scalar_field", f"snodl",
-            "--output_dir", f"./bkg/det_ensres/",
-            "--output_file", f"{to_fv3time(self.task_config.bkg_time)}.sfc_data",
-            "--output_mosaic", f"./orog/ens/{self.task_config.CASE_ENS}_mosaic.nc",
-            "--interp_method", f"conserve_order1",
-            "--weight_file", f"./orog/det/{self.task_config.CASE}.mx{self.task_config.OCNRES}_interp_weight",
-            "--weight_field", f"lsm_frac",
-            "--remap_file", f"./remap",
-        ]
-        fregrid_exe = os.path.join(self.task_config.DATA, 'fregrid.x')
-        exec_cmd = Executable(fregrid_exe)
+        # create a temporary dict of all keys needed in this method
+        localconf = AttrDict()
+        keys = ['DATA', 'current_cycle', 'COMIN_OBS',
+                'OPREFIX', 'CASE', 'OCNRES', 'ntiles', 'FIXgfs']
+        for key in keys:
+            localconf[key] = self.task_config[key]
+
+        localconf['ims_fcst_path'] = './bkg/ensmean/'
+        # Read and render the IMS_OBS_LIST yaml
+        logger.info(f"Reading {self.task_config.IMS_OBS_LIST}")
+        prep_ims_config = parse_j2yaml(self.task_config.IMS_OBS_LIST, localconf)
+        logger.debug(f"{self.task_config.IMS_OBS_LIST}:\n{pformat(prep_ims_config)}")
+
+        # copy the IMS obs files from COMIN_OBS to DATA/obs
+        logger.info("Copying IMS obs for CALCFIMSEXE")
+        FileHandler(prep_ims_config.calcfims).sync()
+
+        logger.info("Create namelist for CALCFIMSEXE")
+        nml_template = self.task_config.FIMS_NML_TMPL
+        nml_data = Jinja(nml_template, localconf).render
+        logger.debug(f"fims.nml:\n{nml_data}")
+
+        nml_file = os.path.join(localconf.DATA, "fims.nml")
+        with open(nml_file, "w") as fho:
+            fho.write(nml_data)
 
+        logger.info("Link CALCFIMSEXE into DATA/")
+        exe_src = self.task_config.CALCFIMSEXE
+        exe_dest = os.path.join(localconf.DATA, os.path.basename(exe_src))
+        if os.path.exists(exe_dest):
+            rm_p(exe_dest)
+        os.symlink(exe_src, exe_dest)
```

Comment:
This should be a copy to meet NCO guidelines.

Reply:
known and noted, we have to go back and make a lot of links copies when the JEDI executables can handle this, this case here a copy should work though

Reply:
Yeah, if it works there's no reason not to do it correctly on all the new stuff so there is less to go back and fix later.